File: src/com/gmail/marcosav2010/communicator/module/fth/packet/PacketGotFile.java
package com.gmail.marcosav2010.communicator.module.fth.packet;

import java.io.IOException;

import com.gmail.marcosav2010.communicator.packet.Packet;
import com.gmail.marcosav2010.communicator.packet.wrapper.PacketDecoder;
import com.gmail.marcosav2010.communicator.packet.wrapper.PacketEncoder;

import lombok.AllArgsConstructor;
import lombok.Getter;

@AllArgsConstructor
public class PacketGotFile extends Packet {

	@Getter
	private String fileName;
	@Getter
	private String owner;

	@Override
	protected void encodeContent(PacketEncoder out) throws IOException {
		out.write(fileName);
		out.write(owner);
	}

	@Override
	protected void decodeContent(PacketDecoder in) throws IOException {
		fileName = in.readString();
		owner = in.readString();
	}
}

File: modules/file-transfer/src/main/java/com/gmail/marcosav2010/fth/packet/PacketGotFile.kt
package com.gmail.marcosav2010.fth.packet

import com.gmail.marcosav2010.communicator.packet.Packet
import com.gmail.marcosav2010.communicator.packet.wrapper.PacketDecoder
import com.gmail.marcosav2010.communicator.packet.wrapper.PacketEncoder
import kotlin.properties.Delegates

class PacketGotFile() : Packet() {

    var fileName by Delegates.notNull<String>()
        private set

    var owner by Delegates.notNull<String>()
        private set

    constructor(fileName: String, owner: String) : this() {
        this.fileName = fileName
        this.owner = owner
    }

    override fun encodeContent(out: PacketEncoder) {
        out.write(fileName)
        out.write(owner)
    }

    override fun decodeContent(`in`: PacketDecoder) {
        fileName = `in`.readString()
        owner = `in`.readString()
    }
}